docs/user-guide/transformations/pivot.md

```diff
@@ -1,17 +1,71 @@
 # Pivots
 
-Pivot a column in a `DataFrame` and perform one of the following aggregations:
+`pivot` transforms a "long-format" `DataFrame`, where each row represents an
+observation, into a "wide-format" one, where each element represents an
+observation.
 
-- first
-- sum
-- min
-- max
-- mean
-- median
+To perform a pivot, specify one or more columns for each of `values`, `index`,
+and `columns`, either by name or via selectors. Typically, the columns in
+`values`, `index`, and `columns` are mutually exclusive; specifying overlapping
+columns will not give an error, but is rarely useful.
 
-The pivot operation consists of a group by one, or multiple columns (these will be the
-new y-axis), the column that will be pivoted (this will be the new x-axis) and an
-aggregation.
+In the simplest case where `values`, `index` and `columns` are each a single
+column:
+
+- Each unique value of the `index` column will become the name of a row in
+  the pivoted `DataFrame`. The first column of the pivoted `DataFrame` will
+  contain these row names.
+- Each unique value of the `columns` column will become the name of a column
+  in the pivoted `DataFrame`.
+- Each value of the `values` column will become a value in the pivoted
+  `DataFrame`. For instance, if the nth row of the input `DataFrame` is
+  `("values_n", "index_n", "columns_n")`, then the value `"values_n"` will
+  be placed at row `"index_n"` (i.e. the row where the `index` column has
+  the value `index_n`) and column `"columns_n"`.
+
+Thus, if there are `N` unique values in the `columns` column, there will be
+`N + 1` columns in the pivoted `DataFrame`: one for the row names, the
+remaining `N` for the values.
```

Comment:
I think this is only true if there's a single `index` column? Which is fine, you explain what happens with multiple index columns below, I just think it needs caveating here